plugins/collections/get.js

```diff
@@ -6,6 +6,42 @@ const schema = require('screwdriver-data-schema');
 const getSchema = schema.models.collection.get;
 const idSchema = joi.reach(schema.models.collection.base, 'id');
 
+/**
+ * Helper function to get last builds of pipelines
+ *
+ * @param   {Array}    pipelines     - array of pipeline models to get last builds for
+ * @param   {Factory}  eventFactory  - factory for getting last event model
+ * @returns {Array}
+ */
+function getPipelinesHealth(pipelines, eventFactory) {
+    return pipelines
+        .filter(pipeline => !!pipeline)
+        .map((pipeline) => {
+            const result = Object.assign({}, pipeline.toJson());
+
+            if (!pipeline.lastEventId) {
+                return result;
+            }
+
+            return eventFactory.get(pipeline.lastEventId)
+                .then((event) => {
+                    if (!event) {
+                        return result;
```

Comment:
If this throws an error here it will be caught by the catch block below which currently returns `result`. I wrote my thought process behind returning the result and not throwing an error instead in a [reply](https://github.com/screwdriver-cd/screwdriver/pull/714#discussion_r134926676) to your comment